org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java

```diff
@@ -11,92 +11,180 @@
+			xLzValue.setFillValue(fill);
+			yLzValue.setFillValue(fill);
+			xLzValue.setChunking(info.createChunk(false, 8)); // Might be slow, need to check this
+			yLzValue.setChunking(info.createChunk(false, 8)); // Might be slow, need to check this
+			xLzValue.setWritingAsync(true);
+			yLzValue.setWritingAsync(true);
+		}
+
+		registerAttributes(positioner, this);
+
+		NexusObjectWrapper<NXslit> nexusDelegate = new NexusObjectWrapper<>(
+				getName(), positioner, NXslit.NX_X_GAP);
+		nexusDelegate.setDefaultAxisDataFieldName(NXslit.NX_X_GAP);
+		nexusDelegate.setCategory(NexusBaseClass.NX_INSTRUMENT);
+		return nexusDelegate;
+	}
+
+	public void setPosition(Number value, IPosition position) throws Exception {
+
+		if (value!=null) {
+			int index = position!=null ? position.getIndex(getName()) : -1;
+			if (isRealisticMove()) {
+				value = doRealisticMove(value, index, -1);
+			}
+			this.position = value;
+			delegate.firePositionPerformed(-1, new Scalar(getName(), index, value.doubleValue()));
+		}
+
+		if (position!=null) {
+			write(value, getPosition(), position);
+		}
+	}
+
+	private void write(Number demand, Number actual, IPosition loc) throws Exception {
+
+		if (xLzValue==null || yLzValue==null) {
+			return;
+		}
+		if (actual!=null) {
+			// write actual position
+			final Dataset newActualPositionData = DatasetFactory.createFromObject(actual);
+			IScanSlice rslice = IScanRankService.getScanRankService().createScanSlice(loc);
+			SliceND xSliceND = new SliceND(xLzValue.getShape(), xLzValue.getMaxShape(), rslice.getStart(), rslice.getStop(), rslice.getStep());
+			SliceND ySliceND = new SliceND(yLzValue.getShape(), yLzValue.getMaxShape(), rslice.getStart(), rslice.getStop(), rslice.getStep());
+			if (isWritingOn()) {
+				xLzValue.setSlice(null, newActualPositionData, xSliceND);
+				yLzValue.setSlice(null, newActualPositionData, ySliceND);
+			}
+		}
+
+		if (xLzSet==null || yLzSet==null) {
+			return;
+		}
+		if (demand!=null) {
+			int index = loc.getIndex(getName());
+			if (index<0) {
+				throw new Exception("Incorrect data index for scan for value of '"+getName()+"'. The index is "+index);
+			}
+			final int[] startPos = new int[] { index };
+			final int[] stopPos = new int[] { index + 1 };
+
+			// write demand position
+			final Dataset newDemandPositionData = DatasetFactory.createFromObject(demand);
+			if (isWritingOn()) {
+				xLzSet.setSlice(null, newDemandPositionData, startPos, stopPos, null);
+				yLzSet.setSlice(null, newDemandPositionData, startPos, stopPos, null);
+			}
+		}
 	}
 
-	public NexusObjectProvider<NXpositioner> getNexusProvider(NexusScanInfo info) throws NexusException {
-        // TODO FIXME Use NeXus API to create slit information.
-		return super.getNexusProvider(info);
+	/**
+	 * Add the attributes for the given attribute container into the given nexus object.
+	 * @param positioner
+	 * @param container

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "registerAttributes" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)